aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/UserAgentDecorator.kt

```diff
@@ -45,9 +49,17 @@ class UserAgentDecorator : ClientCodegenDecorator {
         operation: OperationShape,
         baseCustomizations: List<OperationCustomization>,
     ): List<OperationCustomization> {
-        return baseCustomizations + UserAgentFeature(codegenContext)
+        return baseCustomizations + UserAgentMutateOpRequest(codegenContext)
     }
 
+    override fun serviceRuntimePluginCustomizations(
+        codegenContext: ClientCodegenContext,
+        baseCustomizations: List<ServiceRuntimePluginCustomization>,
+    ): List<ServiceRuntimePluginCustomization> =
+        baseCustomizations.letIf(codegenContext.settings.codegenConfig.enableNewSmithyRuntime) {
+            it + listOf(AddApiMetadataIntoConfigBag(codegenContext))
+        }
+
```

Comment:
could we just gate the usage of this on the flag and always emit the customizations? Also, do we want service plugins to be able to delete others? if not, we could refactor this API to remove baseCustomizations

Reply:
> could we just gate the usage of this on the flag and always emit the customizations?

This seems like bikeshedding to me, but I might be misunderstanding.

> Also, do we want service plugins to be able to delete others?

Hard to say right now. Probably not?